.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,48 @@
+name: build and test
+
+on:
+  push:
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build-and-test:
+    name: build-and-test-${{matrix.os}}
+    runs-on: ${{ matrix.os }}
+    if: github.repository_owner == 'SubnauticaNitrox'
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest, macOS-latest]
+    
+    env:
+      DOTNET_SKIP_FIRST_TIME_EXPERIENCE: true
+      NUGET_PACKAGES: ${{ github.workspace }}/.nuget/package
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+
+    - name: Install BinaryPack Dependencies
+      run: dotnet restore ./src/BinaryPack/BinaryPack.csproj
+
+    - name: Install BinaryPack.Unit Dependencies
+      run: dotnet restore ./unit/BinaryPack.Unit/BinaryPack.Unit.csproj
+
+    - name: Install BinaryPack.Unit.Internals Dependencies
+      run: dotnet restore ./unit/BinaryPack.Unit.Internals/BinaryPack.Unit.Internals.csproj
```

Comment:
Minor: I think would be nicer to maintain as just `dotnet restore`

Reply:
That was on purpose so we don't fetch the packages from the benchmark and comparison projects

Reply:
Fair enough. `dotnet restore` doesn't seem to have an option to run on all projects in a sub-directory.